addons/web/models/ir_http.py

```diff
@@ -137,21 +137,38 @@ def _get_content_common(self, xmlid=None, model='ir.attachment', res_id=None, fi
             xmlid=xmlid, model=model, id=res_id, field=field, unique=unique, filename=filename,
             filename_field=filename_field, download=download, mimetype=mimetype, access_token=access_token
         )
+        placeholder_content = None
         if status != 200:
-            return self._response_by_status(status, headers, content)
-        else:
-            headers.append(('Content-Length', len(content)))
-            response = request.make_response(content, headers)
+            if filename and filename.endswith(".mp4") and not content:
+                # Show placeholder video if video should've 404'd
+                placeholder_content = self._video_placeholder()
+                headers.append(('Content-Type', "video/mp4"))
+            else:
+                return self._response_by_status(status, headers, content)
+
+        content = placeholder_content or (content if not isinstance(content, tuple) else "")
+        headers.append(('Content-Length', len(content)))
+
+        # Setting CSP required in order to play same origin videos as BG videos.
```

Comment:
What BG videos?

Reply:
If you set a video that was uploaded to Odoo as the background of a snippet, it requires ('Content-Security-Policy', "media-src 'self'") in order to play them.